src/time_zone_impl.cc

```diff
@@ -45,7 +45,8 @@ time_zone time_zone::Impl::UTC() {
   return time_zone(UTCImpl());
 }
 
-bool time_zone::Impl::LoadTimeZone(const std::string& name, time_zone* tz) {
+bool time_zone::Impl::LoadTimeZone(detail::char_range name_range, time_zone* tz) {
```

Comment:
Given that these interfaces immediately turn around and create a `std::string` (at least for the foreseeable future), there doesn't seem to be much point in modifying the `LoadTimeZone()` path.

Indeed, allow me to restate an earlier question:  Is there evidence that we're spending significant amounts of time converting non-`std::string`s to `std::string`s such that we need to push that non-conversion through.  (Naively, I'd be surprised if that conversion was the dominating cost of formatting/parsing.)